cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/DefaultStore.java

```diff
@@ -329,6 +330,27 @@ public int countActiveRuns(@Nullable Integer limit) {
                                   context -> (int) getAppMetadataStore(context).countActiveRuns(limit));
   }
 
+  @Override
+  public void scanActiveRuns(int txBatchSize, Consumer<RunRecordDetail> consumer) {
+    AtomicInteger count = new AtomicInteger();
+    AtomicReference<AppMetadataStore.Cursor> cursorRef = new AtomicReference<>(AppMetadataStore.Cursor.EMPTY);
+
+    while (true) {
+      TransactionRunners.run(transactionRunner, context -> {
+        AppMetadataStore store = AppMetadataStore.create(context);
+        store.scanActiveRuns(cursorRef.get(), txBatchSize, (cursor, runRecordDetail) -> {
```

Comment:
In the previous implementation we had a retry policy here. Do we need it?

Reply:
Done!